### PR TITLE
Improvement/pro 1192 new tx history

### PR DESCRIPTION
**Pull request by bhumitaylor14**
## Description
Display corrrect gas fee as well as token value

## Motivation and Context
Remove transactions which have null data of asssets and display correct token value and gas fee for all txs

## How Has This Been Tested?
Tested on my local machine

## Screenshots (if appropriate):
Screenshot 2023-11-10 at 2 11 48 PM

## Types of changes

- [ tick] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/components/TransactionPreview/HistoryPreview.tsx [hunk 1]
```diff
@@ -128,9 +121,17 @@ const HistoryPreview = ({ crossChainAction }: TransactionPreviewInterface) => {
               />
             </ClickableText>
             <ClickableText onClick={handleTap}>
-              <Text size={12} marginBottom={2} marginRight={3} medium block>
-                <LuFuel size={16} style={{ marginRight: 5 }} color={theme?.color?.text?.innerLabel} />
-                {showFullGasAmount ? <Text> {fullGasFeesNativeToken}</Text> : <Text> {gasFeesNativeToken}</Text>}
+              <Text size={16} marginBottom={2} marginRight={5} medium block>
+                <LuFuel size={16} style={{ marginRight: 8 }} color={theme?.color?.text?.innerLabel} />
+                {parseFloat(gasfeesInUsd.toFixed(3)) < 0.001 ? (
```
Comment (poocart) [inline src/components/TransactionPreview/HistoryPreview.tsx:126]: please avoid nested ternary

Comment (bhumitaylor14) [reply src/components/TransactionPreview/HistoryPreview.tsx]: changes done.

File: src/components/TransactionPreview/HistoryPreview.tsx [hunk 2]
```diff
@@ -66,32 +66,25 @@ const HistoryPreview = ({ crossChainAction }: TransactionPreviewInterface) => {
...
+    const gasfeeDisplay = `${partialgasfeesInUsd}  $`;
 
-    const gasFeesUSD = `${gasFees} $`;
-    const gasFeesNativeToken = `${partialgasFees + ` ` + asset.symbol}`;
-    console.log(gasFeesNativeToken);
-    const fullGasFeesNativeToken = `${finalgasFees + ` ` + asset.symbol}`;
-    console.log(fullGasFeesNativeToken);
+    // calculate Token value
+    const tokenValue =
+      asset.feeAmount && asset.feeAmount != null ? ethers.utils.formatUnits(asset?.feeAmount, asset?.decimals) : null;
```
Comment (poocart) [inline src/components/TransactionPreview/HistoryPreview.tsx:80]:
```suggestion
      asset.feeAmount && asset.feeAmount != null ? ethers.utils.formatUnits(asset.feeAmount, asset.decimals) : null;
```

File: src/components/TransactionPreview/HistoryPreview.tsx [hunk 3]
```diff
@@ -66,32 +66,25 @@ const HistoryPreview = ({ crossChainAction }: TransactionPreviewInterface) => {
...
-
-    // convert Tx fee to USD
-    let transactionfeesInUsd = transactionfees * crossChainAction.estimated.usdPrice;
-    transactionfeesInUsd =
-      transactionfeesInUsd.toFixed(2) == '0.00' ? transactionfeesInUsd.toFixed(6) : transactionfeesInUsd.toFixed(2);
-
-    const amountString = `${+transactionfees + ` ` + asset.symbol}・ $ ${transactionfeesInUsd}`;
+    // convert Token value to USD
+    const tokenValueUsd =
+      tokenValue != null ? formatAmountDisplay(`${+tokenValue * +crossChainAction.estimated.usdPrice}`, '$') : '0 $';
```
Comment (poocart) [inline src/components/TransactionPreview/HistoryPreview.tsx:84]:
```suggestion
      tokenValue != null ? formatAmountDisplay(`${+tokenValue * +crossChainAction.estimated.usdPrice}`, '$') : '$0';
```

Comment (poocart) [reply src/components/TransactionPreview/HistoryPreview.tsx:84]: to match what `formatAmountDisplay` returns

File: src/components/History/History.tsx [hunk 4]
```diff
@@ -48,67 +48,69 @@ const History = ({ onBackButtonClick }: { onBackButtonClick: () => void }) => {
...
-            fromAddress,
-            isFromEtherspotWallet,
-            transactionUrl,
-            asset: {
-              address: item.asset ? item.asset.from : item.from,
-              decimals: item.asset ? item.asset.decimal : 2,
-              symbol: item.asset ? item.asset.symbol : '',
-              amount: '100000000000000000',
-              iconUrl: assetnetwork != null ? assetnetwork?.logoURI : '',
+          if (item.asset != null) {
```
Comment (poocart) [inline src/components/History/History.tsx:51]:
```suggestion
          if (!item?.asset) return;
```

and then no need to wrap rest of the code inside operator brackets

File: src/components/History/History.tsx [hunk 5]
```diff
@@ -48,67 +48,69 @@ const History = ({ onBackButtonClick }: { onBackButtonClick: () => void }) => {
...
+            const assetnetwork =
```
Comment (poocart) [inline src/components/History/History.tsx:52]: please follow camelCase naming standards

```suggestion
            const assetNetwork =
```

File: src/components/History/History.tsx [hunk 6]
```diff
@@ -48,67 +48,69 @@ const History = ({ onBackButtonClick }: { onBackButtonClick: () => void }) => {
...
+                useWeb3Provider: !isFromEtherspotWallet,
+                direction: item.direction,
+              },
+            ];
+
+            storedTransactionsDetails[crossChainActionId] = [...crossChainAction];
+
+            const storedGroupedCrossChainActionsUpdated = storedTransactionsDetails;
+
+            setStoredGroupedCrossChainActions(storedGroupedCrossChainActionsUpdated);
```
Comment (poocart) [inline src/components/History/History.tsx:112]:
```suggestion
            setStoredGroupedCrossChainActions(storedTransactionsDetails);
```

File: src/components/TransactionPreview/HistoryPreview.tsx [hunk 7]
```diff
@@ -66,32 +66,25 @@ const HistoryPreview = ({ crossChainAction }: TransactionPreviewInterface) => {
     // Convert gas wei to ETH
     const gasInETHAmount = gasCost / 1e18;
 
-    // Convert to USD
-    let gasFees = gasInETHAmount * crossChainAction.estimated.usdPrice;
-    let gasFeesNative = gasInETHAmount;
+    // calculate gas fees
+    const gasfees = gasInETHAmount * asset.gasUsed;
```
Comment (poocart) [inline src/components/TransactionPreview/HistoryPreview.tsx:70]: please follow camelCase naming standards

```suggestion
    const gasFees = gasInETHAmount * asset.gasUsed;
```

File: src/components/TransactionPreview/HistoryPreview.tsx [hunk 8]
```diff
@@ -66,32 +66,25 @@ const HistoryPreview = ({ crossChainAction }: TransactionPreviewInterface) => {
...
 
-    gasFeesNative = parseFloat(gasInETHAmount.toFixed(4)) < 0.0001 ? '<0.0001' : gasInETHAmount.toFixed(5);
-    const partialgasFees = gasFeesNative.toString();
-    let fullGasFees = gasInETHAmount.toFixed(15);
-    const finalgasFees = fullGasFees.toString();
-    console.log(fullGasFees);
+    // convert gas fee to USD
+    const gasfeesInUsd = gasfees * crossChainAction.estimated.usdPrice;
```
Comment (poocart) [inline src/components/TransactionPreview/HistoryPreview.tsx:73]: please follow camelCase naming standards

```suggestion
    const gasFeesInUsd = gasfees * crossChainAction.estimated.usdPrice;
```

File: src/components/TransactionPreview/HistoryPreview.tsx [hunk 9]
```diff
@@ -66,32 +66,25 @@ const HistoryPreview = ({ crossChainAction }: TransactionPreviewInterface) => {
...
+    const fullgasFee = gasfeesInUsd.toFixed(6);
```
Comment (poocart) [inline src/components/TransactionPreview/HistoryPreview.tsx:74]: please follow camelCase naming standards

```suggestion
    const fullGasFee = gasfeesInUsd.toFixed(6);
```

File: src/components/TransactionPreview/HistoryPreview.tsx [hunk 10]
```diff
@@ -66,32 +66,25 @@ const HistoryPreview = ({ crossChainAction }: TransactionPreviewInterface) => {
...
+    const partialgasfeesInUsd = parseFloat(gasfeesInUsd.toFixed(3)) < 0.001 ? '<0.001' : gasfeesInUsd.toFixed(2);
```
Comment (poocart) [inline src/components/TransactionPreview/HistoryPreview.tsx:75]: please follow camelCase naming standards

```suggestion
    const partialGasFeesInUsd = parseFloat(gasfeesInUsd.toFixed(3)) < 0.001 ? '<0.001' : gasfeesInUsd.toFixed(2);
```

File: src/components/TransactionPreview/HistoryPreview.tsx [hunk 11]
```diff
@@ -66,32 +66,25 @@ const HistoryPreview = ({ crossChainAction }: TransactionPreviewInterface) => {
...
-    transactionfeesInUsd =
-      transactionfeesInUsd.toFixed(2) == '0.00' ? transactionfeesInUsd.toFixed(6) : transactionfeesInUsd.toFixed(2);
-
-    const amountString = `${+transactionfees + ` ` + asset.symbol}・ $ ${transactionfeesInUsd}`;
+    // convert Token value to USD
+    const tokenValueUsd =
+      tokenValue != null ? formatAmountDisplay(`${+tokenValue * +crossChainAction.estimated.usdPrice}`, '$') : '0 $';
+    const tokenfeeDisplay = `${direction === 'Sender' ? '-' : '+'} ${
+      +tokenValue + ` ` + asset.symbol
+    }・${tokenValueUsd}`;
```
Comment (poocart) [inline src/components/TransactionPreview/HistoryPreview.tsx:87]:
```suggestion
    const tokenfeeDisplay = `${direction === 'Sender' ? '-' : '+'} ${tokenValue}  ${asset.symbol}・${tokenValueUsd}`;
```

File: src/components/TransactionPreview/HistoryPreview.tsx [hunk 12]
```diff
@@ -66,32 +66,25 @@ const HistoryPreview = ({ crossChainAction }: TransactionPreviewInterface) => {
...
-    gasFeesNative = parseFloat(gasInETHAmount.toFixed(4)) < 0.0001 ? '<0.0001' : gasInETHAmount.toFixed(5);
-    const partialgasFees = gasFeesNative.toString();
-    let fullGasFees = gasInETHAmount.toFixed(15);
-    const finalgasFees = fullGasFees.toString();
-    console.log(fullGasFees);
+    // convert gas fee to USD
+    const gasfeesInUsd = gasfees * crossChainAction.estimated.usdPrice;
+    const fullgasFee = gasfeesInUsd.toFixed(6);
+    const partialgasfeesInUsd = parseFloat(gasfeesInUsd.toFixed(3)) < 0.001 ? '<0.001' : gasfeesInUsd.toFixed(2);
+    const gasfeeDisplay = `${partialgasfeesInUsd}  $`;
```
Comment (poocart) [inline src/components/TransactionPreview/HistoryPreview.tsx:76]: please follow camelCase naming standards

```suggestion
    const gasFeeDisplay = `${partialgasfeesInUsd}  $`;
```

File: src/components/TransactionPreview/HistoryPreview.tsx [hunk 13]
```diff
@@ -128,9 +121,13 @@ const HistoryPreview = ({ crossChainAction }: TransactionPreviewInterface) => {
...
-              <Text size={12} marginBottom={2} marginRight={3} medium block>
-                <LuFuel size={16} style={{ marginRight: 5 }} color={theme?.color?.text?.innerLabel} />
-                {showFullGasAmount ? <Text> {fullGasFeesNativeToken}</Text> : <Text> {gasFeesNativeToken}</Text>}
+              <Text size={16} marginBottom={2} marginRight={5} medium block>
+                <LuFuel size={16} style={{ marginRight: 8 }} color={theme?.color?.text?.innerLabel} />
+                {parseFloat(gasfeesInUsd.toFixed(3)) < 0.001 && showFullGasAmount ? (
+                  <Text> {fullgasFee}</Text>
+                ) : (
+                  <Text> {gasfeeDisplay}</Text>
+                )}
```
Comment (poocart) [inline src/components/TransactionPreview/HistoryPreview.tsx:130]:
```suggestion
                <Text> {parseFloat(gasfeesInUsd.toFixed(3)) < 0.001 && showFullGasAmount ? fullgasFee : gasfeeDisplay}
                </Text>
```

File: src/components/History/History.tsx [hunk 14]
```diff
@@ -34,7 +34,7 @@ const History = ({ onBackButtonClick }: { onBackButtonClick: () => void }) => {
   //pull all Transaction data and set in array to display on UI
   const pullAllNewTx = async (accountAddress: string | null) => {
     setIsLoadingTransactions(true);
-    const transactions = await getAllTransactions(accountAddress);
+    const transactions = await getAllTransactions('0x89a3d6AF00a3627DA25E2e8FFCCb97FE74D52631');
```
Comment (poocart) [inline src/components/History/History.tsx:37]: what is this address?

```suggestion
    const transactions = await getAllTransactions('0x89a3d6AF00a3627DA25E2e8FFCCb97FE74D52631');
```

File: src/components/History/History.tsx [hunk 15]
```diff
@@ -106,9 +107,9 @@ const History = ({ onBackButtonClick }: { onBackButtonClick: () => void }) => {
 
           storedTransactionsDetails[crossChainActionId] = [...crossChainAction];
 
-          const storedGroupedCrossChainActionsUpdated = storedTransactionsDetails;
+          //const storedGroupedCrossChainActionsUpdated = storedTransactionsDetails;
```
Comment (poocart) [inline src/components/History/History.tsx:110]:
```suggestion
```

File: src/components/TransactionPreview/HistoryPreview.tsx [hunk 16]
```diff
@@ -66,32 +66,23 @@ const HistoryPreview = ({ crossChainAction }: TransactionPreviewInterface) => {
...
-    gasFeesNative = parseFloat(gasInETHAmount.toFixed(4)) < 0.0001 ? '<0.0001' : gasInETHAmount.toFixed(5);
-    const partialgasFees = gasFeesNative.toString();
-    let fullGasFees = gasInETHAmount.toFixed(15);
-    const finalgasFees = fullGasFees.toString();
-    console.log(fullGasFees);
+    // convert gas fee to USD
+    const gasFeesInUsd = gasFees * crossChainAction.estimated.usdPrice;
+    const fullGasFee = gasFeesInUsd.toFixed(6);
+    const partialGasFeesInUsd = parseFloat(gasFeesInUsd.toFixed(3)) < 0.001 ? '<0.001' : gasFeesInUsd.toFixed(2);
+    const gasFeeDisplay = `${partialGasFeesInUsd}  $`;
```
Comment (poocart) [inline src/components/TransactionPreview/HistoryPreview.tsx:76]:
```suggestion
    const gasFeeDisplay = `#${partialGasFeesInUsd}`;
```

File: src/components/TransactionPreview/HistoryPreview.tsx [hunk 17]
```diff
@@ -66,32 +66,23 @@ const HistoryPreview = ({ crossChainAction }: TransactionPreviewInterface) => {
...
-
-    // convert Tx fee to USD
-    let transactionfeesInUsd = transactionfees * crossChainAction.estimated.usdPrice;
-    transactionfeesInUsd =
-      transactionfeesInUsd.toFixed(2) == '0.00' ? transactionfeesInUsd.toFixed(6) : transactionfeesInUsd.toFixed(2);
-
-    const amountString = `${+transactionfees + ` ` + asset.symbol}・ $ ${transactionfeesInUsd}`;
+    // convert Token value to USD
+    const tokenValueUsd =
+      tokenValue != null ? formatAmountDisplay(`${+tokenValue * +crossChainAction.estimated.usdPrice}`, '$') : '$ 0';
```
Comment (poocart) [inline src/components/TransactionPreview/HistoryPreview.tsx:84]:
```suggestion
      tokenValue != null ? formatAmountDisplay(`${+tokenValue * +crossChainAction.estimated.usdPrice}`, '$') : '$0';
```

File: src/components/History/History.tsx [hunk 18]
```diff
@@ -48,7 +48,8 @@ const History = ({ onBackButtonClick }: { onBackButtonClick: () => void }) => {
         const assets = await getSupportedAssetsForChainId(chain_id);
 
         value.map((item) => {
-          const assetnetwork =
+          if (!item?.asset) return;
```
Comment (poocart) [inline src/components/History/History.tsx:51]:
```suggestion
          if (!item?.asset) {
            // add some log maybe, did u manage to catch this case?
            return;
          }
```

**Review by poocart — APPROVED** [review]
lgtm

**Pull request merged**